File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/c_solutions/3090-maximum-length-substring-with-two-occurrences.c
int maximumLengthSubstring(char* s) {
    int count[26] = {0};
    int left = 0, right = 0;
    int maxLen = 0;

    while (s[right] != '\0') {
        count[s[right] - 'a']++;

        while (count[s[right] - 'a'] > 2) {
            count[s[left] - 'a']--;
            left++;
        }

        maxLen = (right - left + 1) > maxLen ? (right - left + 1) : maxLen;
        right++;
    }

    return maxLen;
}

File: leetcode/AIcoding/3.DeepSeek-v3/python_solutions/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        max_len = 0
        left = 0
        count = {}

        for right in range(len(s)):
            char = s[right]
            count[char] = count.get(char, 0) + 1

            while count[char] > 2:
                left_char = s[left]
                count[left_char] -= 1
                if count[left_char] == 0:
                    del count[left_char]
                left += 1

            max_len = max(max_len, right - left + 1)

        return max_len